src/core/shader.js

```diff
@@ -96,8 +95,22 @@ Shader.prototype = {
     var schema = this.schema;
     dataKeys.forEach(processData);
     function processData (key) {
+      var materialKey;
       if (!schema[key] || schema[key].is !== type) { return; }
-      if (variables[key].value === data[key]) { return; }
+      if (schema[key].type === 'map') {
+        // Special handling is needed for textures.
+        materialKey = '_texture_' + key;
+        // If data unchanged, get out early.
+        if (!variables[key] || variables[key].value === data[key]) { return; }
+        // We can't actually set the variable correctly until we've loaded the texture.
+        self.el.addEventListener('materialtextureloaded', function (e) {
+          variables[key].value = self.material[materialKey];
+          variables[key].needsUpdate = true;
+        });
+        // Kick off the texture update now that handler is added.
+        utils.material.updateMapMaterialFromData(materialKey, key, self, data);
```

Comment:
sure, but I am not sure what texture loader you are asking to use so I am not sure how to execute, is what I'm saying.